virtcontainers/pkg/oci/utils.go

```diff
@@ -570,7 +546,9 @@ func ContainerConfig(ocispec CompatOCISpec, bundlePath, cid, console string, det
 	}
 	if ocispec.Linux.Resources.Memory != nil {
 		if ocispec.Linux.Resources.Memory.Limit != nil {
-			resources.MemMB = uint32(*ocispec.Linux.Resources.Memory.Limit >> 20)
+			// do page align to memory, as cgroup memory.limit_in_bytes will be aligned to page when effect
+			// TODO use GetGuestDetails to get the guest OS page size.
+			resources.MemByte = (*ocispec.Linux.Resources.Memory.Limit >> 12) << 12
```

Comment:
same here. Please create a helper function and use `os.Getpagesize()` for it.

Reply:
Yes, we have talked about this before that we should get guest page size here, so I have added a `TODO use GetGuestDetails to get the guest OS page size.`

Reply:
You mean we use os.Getpagesize for now? But the `12` means guest page size here. It's not proper to use `os.Getpagesize()` here.